File: app/src/main/java/com/pcsalt/example/productflavors/utils/MyLog.java
package com.pcsalt.example.productflavors.utils;

import android.util.Log;

/**
 * Created by pcsalt on 27/05/16.
 */
public final class MyLog {

    private MyLog() {
    }

    public static void i(String tag, String message) {
        if (FConsts.ENABLE_LOG) {
            Log.i(tag, message);
        }
    }

    public static void d(String tag, String message) {
        if (FConsts.ENABLE_LOG) {
            Log.d(tag, message);
        }
    }

    public static void w(String tag, String message) {
        if (FConsts.ENABLE_LOG) {
            Log.w(tag, message);
        }
    }

    public static void e(String tag, String message) {
        if (FConsts.ENABLE_LOG) {
            Log.e(tag, message);
        }
    }

    public static void e(String tag, String message, Throwable throwable) {
        if (FConsts.ENABLE_LOG) {
            Log.e(tag, message, throwable);
        }
    }
}

File: app/src/main/kotlin/com/pcsalt/example/productflavors/utils/MyLog.kt
package com.pcsalt.example.productflavors.utils

import android.util.Log

object MyLog {

  fun i(tag: String, message: String) {
    if (FConsts.ENABLE_LOG) Log.i(tag, message)
  }

  fun d(tag: String, message: String) {
    if (FConsts.ENABLE_LOG) Log.d(tag, message)
  }

  fun w(tag: String, message: String) {
    if (FConsts.ENABLE_LOG) Log.w(tag, message)
  }

  fun e(tag: String, message: String) {
    if (FConsts.ENABLE_LOG) Log.e(tag, message)
  }

  fun e(tag: String, message: String, throwable: Throwable) {
    if (FConsts.ENABLE_LOG) Log.e(tag, message, throwable)
  }
}
